pages/api/studentPortal/quizzes/FirstLang.ts

```diff
@@ -0,0 +1,67 @@
+type QuizData = {
+    name?: string;
```

Comment:
this is not good, because all of the properties in this type are optional. You're kind of using this type as a placeholder, than can have any or all of these properties. 

Think about what an instance of a QuizData is conceptually inside of our business. Imagine having multiple quizzes, each with their own quiz data, what makes up a quiz? Think about the exercise we did with creating a design for different components in the system. You want your code to be a mirror of your mental model of quizzes.

Reply:
what if you used this for the QuizData type: 

```
type QuizQuestion = {
  title: string;
  body: string;
  options: QuizOption[];
  maxSelections?: number;
};

type QuizOption = {
  name: string;
  result?: string;
  weight?: number;
};

type QuizData = {
  title: string;
  body: string;
  questions: QuizQuestion[];
};
```